Tribler/Test/Core/Modules/test_versioncheck.py

```diff
@@ -0,0 +1,74 @@
+import json
+from twisted.internet.defer import Deferred
+from twisted.web import server, resource
+from Tribler.Core.Modules import versioncheck_manager
+from Tribler.Core.Utilities.network_utils import get_random_port
+from Tribler.Core.Utilities.twisted_thread import reactor, deferred
+from Tribler.Core.simpledefs import NTFY_NEW_VERSION, NTFY_INSERT
+from Tribler.Test.test_as_server import TestAsServer
+
+
+class VersionResource(resource.Resource):
+
+    def __init__(self, response, response_code):
+        resource.Resource.__init__(self)
+        self.response = response
+        self.response_code = response_code
+
+    isLeaf = True
+
+    def render_GET(self, request):
+        request.setResponseCode(self.response_code)
+        return self.response
+
+
+class TestVersionCheck(TestAsServer):
+
+    def setUp(self):
+        self.port = get_random_port()
+        self.server = None
+        self.should_call_new_version_callback = False
+        self.new_version_called = False
+        self.test_deferred = Deferred()
+        versioncheck_manager.VERSION_CHECK_URL = 'http://localhost:%s' % self.port
+        super(TestVersionCheck, self).setUp()
+
+        self.session.notifier.add_observer(self.new_version_callback, NTFY_NEW_VERSION, [NTFY_INSERT])
+
+    def new_version_callback(self, subject, changeType, objectID, *args):
+        self.new_version_called = True
+
+    def listen(self, response, response_code):
+        site = server.Site(VersionResource(response, response_code))
+        self.server = reactor.listenTCP(self.port, site)
+
+    def setup_version_server(self, response, response_code=200):
+        reactor.callFromThread(self.listen, response, response_code)
+
+    def assert_new_version_called(self):
+        self.assertTrue(self.new_version_called == self.should_call_new_version_callback)
+        stop_deferred = self.server.stopListening()
+        stop_deferred.chainDeferred(self.test_deferred)
+
+    def start_version_checker(self):
+        self.session.lm.version_check_manager.start(10000)
+        reactor.callLater(0.5, self.assert_new_version_called)
```

Comment:
Don't use delayed calls for this, each test is adding half a second of runtime and it can also break if the system where the tests run are under load.

See how to simulate the passing of time on the leveldb store tests.